src/main/java/org/sagebionetworks/bridge/spring/controllers/SharedAssessmentController.java

```diff
@@ -0,0 +1,168 @@
+    public PagedResourceList<AssessmentDto> getSharedAssessmentRevisionsByGuid(
+            @PathVariable String guid,
+            @RequestParam(required = false) String offsetBy, 
+            @RequestParam(required = false) String pageSize,
+            @RequestParam(required = false) String includeDeleted) {
+        
+        int offsetByInt = BridgeUtils.getIntOrDefault(offsetBy, 0);
+        int pageSizeInt = BridgeUtils.getIntOrDefault(pageSize, API_DEFAULT_PAGE_SIZE);
+        boolean incDeletedBool = Boolean.valueOf(includeDeleted);
+        
+        PagedResourceList<Assessment> page = service.getAssessmentRevisionsByGuid(
+                SHARED_STUDY_ID_STRING, guid, offsetByInt, pageSizeInt, incDeletedBool);
+
+        List<AssessmentDto> dtos = page.getItems().stream()
+                .map(assessment -> AssessmentDto.create(assessment))
+                .collect(Collectors.toList());
+        
+        return new PagedResourceList<>(dtos, page.getTotal())
+                .withAllRequestParams(page.getRequestParams());
+    }
+
+    @GetMapping("/v1/sharedassessments/identifier:{identifier}/revisions")
+    public PagedResourceList<AssessmentDto> getSharedAssessmentRevisionsById(
+            @PathVariable String identifier,
+            @RequestParam(required = false) String offsetBy, 
+            @RequestParam(required = false) String pageSize,
+            @RequestParam(required = false) String includeDeleted) {
+        
+        int offsetByInt = BridgeUtils.getIntOrDefault(offsetBy, 0);
+        int pageSizeInt = BridgeUtils.getIntOrDefault(pageSize, API_DEFAULT_PAGE_SIZE);
+        boolean incDeletedBool = Boolean.valueOf(includeDeleted);
+        
+        PagedResourceList<Assessment> page = service.getAssessmentRevisionsById(
+                SHARED_STUDY_ID_STRING, identifier, offsetByInt, pageSizeInt, incDeletedBool);
+
+        List<AssessmentDto> dtos = page.getItems().stream()
+                .map(assessment -> AssessmentDto.create(assessment))
+                .collect(Collectors.toList());
+        
+        return new PagedResourceList<>(dtos, page.getTotal())
+                .withAllRequestParams(page.getRequestParams());
+    }
+
+    @PostMapping("/v1/sharedassessments/{guid}")
+    public AssessmentDto updateSharedAssessment(@PathVariable String guid) {
+        UserSession session = getAuthenticatedSession(DEVELOPER);
+        String appId = session.getStudyIdentifier().getIdentifier();
+        
+        AssessmentDto dto = parseJson(AssessmentDto.class);
+        Assessment assessment = Assessment.create(dto, SHARED_STUDY_ID_STRING);
+        
+        // Note that we are passing in the appId of the caller, and the assessment is in the 
+        // shared app, which is the opposite of all the other shared calls
+        Assessment retValue = service.updateSharedAssessment(appId, assessment);
+        
+        return AssessmentDto.create(retValue);
+    }
```

Comment:
Similarly, do we need an update shared assessment API?

Reply:
Yes, the owners of the shared assessment can update the metadata portion of the assessment (there won't be a way to update the configuration).